test/ngMessages/messagesSpec.js

```diff
@@ -412,6 +412,111 @@ describe('ngMessages', function() {
   }));
 
 
+  it('should not crash or leak memory when the messages are transcluded, the first message is' +
+    'visible, and ngMessages is removed by ngIf', function() {
+
+    module(function($compileProvider) {
+      $compileProvider.directive('messageWrap', function() {
+        return {
+          transclude: true,
+          scope: {
+            col: '=col'
+          },
+          template: '<div ng-messages="col"><ng-transclude></ng-transclude></div>'
+        };
+      });
+    });
+
+    inject(function($rootScope, $compile) {
+
+      element = $compile('<div><div ng-if="show"><div message-wrap col="col">' +
+                         '        <div ng-message="a">A</div>' +
+                         '        <div ng-message="b">B</div>' +
+                         '</div></div></div>')($rootScope);
+
+      $rootScope.$apply(function() {
+        $rootScope.show = true;
+        $rootScope.col = {
+          a: true,
+          b: true
+        };
+      });
+
+      expect(messageChildren(element).length).toBe(1);
+      expect(trim(element.text())).toEqual('A');
+
+      $rootScope.$apply('show = false');
+
+      expect(messageChildren(element).length).toBe(0);
+    });
+  });
+
+
+  it('should not crash when the first of two nested messages is removed', function() {
+    inject(function($rootScope, $compile) {
+
+      element = $compile(
+        '<div ng-messages="col">' +
+          '<div class="wrapper">' +
+            '<div ng-message="a">A</div>' +
+            '<div ng-message="b">B</div>' +
+          '</div>' +
+        '</div>'
+      )($rootScope);
+
+      $rootScope.$apply(function() {
+        $rootScope.col = {
+          a: true,
+          b: false
+        };
+      });
+
+      expect(messageChildren(element).length).toBe(1);
+      expect(trim(element.text())).toEqual('A');
+
+      var nodeA = element[0].querySelector('[ng-message="a"]');
+      nodeA.remove();
+
+      expect(messageChildren(element).length).toBe(0);
+    });
+  });
+
+
+  it('should show deeply nested messages correctly after a message has been removed', function() {
```

Comment:
Actually ... since I converted the tests to remove the element via .remove() directly instead of inside of a watcher, the tests don't fail anymore without the patch. This issue is killing me ...